src/Npgsql/NpgsqlCommand.cs

```diff
@@ -243,7 +243,7 @@ public override UpdateRowSource UpdatedRowSource
         /// Returns whether this query will execute as a prepared (compiled) query.
         /// </summary>
         public bool IsPrepared =>
-            _connectorPreparedOn == _connection?.Connector &&
+            _connectorPreparedOn == (_connection?.Connector  ?? _connector) &&
```

Comment:
Looks good - but do you think this addresses any real scenario? _connector is only non-null when coming from type handlers, which don't prepare, right?

(not against doing this at all)

Reply:
At this point it's for the physical callback (which still uses `NpgsqlConnector`).